airbyte-workers/src/main/java/io/airbyte/workers/temporal/sync/SyncWorkflowImpl.java

```diff
@@ -59,6 +65,23 @@ public StandardSyncOutput run(final JobRunConfig jobRunConfig,
     if (syncInput.getOperationSequence() != null && !syncInput.getOperationSequence().isEmpty()) {
       for (final StandardSyncOperation standardSyncOperation : syncInput.getOperationSequence()) {
         if (standardSyncOperation.getOperatorType() == OperatorType.NORMALIZATION) {
+          final int normalizationSummaryCheckVersion =
+              Workflow.getVersion(NORMALIZATION_SUMMARY_CHECK_TAG, Workflow.DEFAULT_VERSION, NORMALIZATION_SUMMARY_CHECK_CURRENT_VERSION);
+          if (normalizationSummaryCheckVersion >= NORMALIZATION_SUMMARY_CHECK_CURRENT_VERSION) {
+            Boolean shouldRun;
+            try {
+              shouldRun = normalizationSummaryCheckActivity.shouldRunNormalization(Long.valueOf(jobRunConfig.getJobId()), jobRunConfig.getAttemptId(),
+                  Optional.of(syncOutput.getStandardSyncSummary().getTotalStats().getRecordsCommitted()));
+            } catch (final IOException e) {
+              shouldRun = true;
+            }
+            if (!shouldRun) {
+              LOGGER.info("Skipping normalization because there are no records to normalize.");
+              break;
```

Comment:
Are we sure we want to break in this case? I think there may be other steps after normalization, like dbt for example.